datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -406,6 +421,34 @@ impl HashJoinExec {
         JoinSide::Right
     }
 
+    /// Return whether the join contains a projection
+    pub fn contain_projection(&self) -> bool {
+        self.projection.is_some()
+    }
+
+    /// Return new instance of [HashJoinExec] with the given projection.
+    pub fn with_projection(&self, projection: Option<Vec<usize>>) -> Result<Self> {
+        //  check if the projection is valid
+        datafusion_common::utils::can_project(&self.schema, projection.as_ref())?;
```

Comment:
Shouldn't it be can_project against `self.schema()` then? Otherwise "layering" projections over each other could potentially panic at L435 if I'm not mistaken.

Reply:
> Shouldn't it be can_project against `self.schema()` then? Otherwise "layering" projections over each other could potentially panic at L435 if I'm not mistaken.

Right. I neglected  it when I merged the code😭.